src/backend/db/database_manager.py

```diff
@@ -418,5 +420,421 @@ def delete_custom_field(self, customer_guid, field_name):
             session.rollback()
             return {"status": "error", "error": str(e)}
+        finally:
+            session.close()
+
+
+    # Tickets related function
+    def create_ticket(self, customer_guid, chat_id, title, description, priority, custom_fields):
+        customer_db_name = self.get_customer_db(customer_guid)
+        session = DatabaseManager._session_factory()
+
+        try:
+            try:
+                # Check if the database exists
+                logger.debug(f"Checking if database {customer_db_name} exists")
+                result = session.execute(
+                    text("SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = :db_name"),
+                    {"db_name": customer_db_name}
+                ).fetchone()
+            except OperationalError as e:
+                logger.error(f"Database connectivity issue during schema check: {e}")
+                raise e
+
+            if not result:
+                raise ValueError(f"Database {customer_db_name} does not exist")
+
+            # Switch to the customer database
+            logger.debug(f"Switching to customer database: {customer_db_name}")
+            session.execute(text(f"USE `{customer_db_name}`"))
+
+            # Check if the chat_id exists in the chat_messages table
+            chat_exists = session.execute(
+                text("SELECT 1 FROM chat_messages WHERE chat_id = :chat_id LIMIT 1"),
+                {"chat_id": chat_id}
+            ).fetchone()
+
+            if not chat_exists:
+                raise ValueError(f"Invalid chat_id: {chat_id} does not exist.")
+
+            # Validate custom fields against the database columns
+            if custom_fields:
+                # Fetch the valid column names from the custom_field_values table
+                result = session.execute("SHOW COLUMNS FROM custom_field_values").fetchall()
+                valid_columns = {row[0] for row in result}
+
+                # Check for invalid fields
+                invalid_fields = [field for field in custom_fields.keys() if field not in valid_columns]
+                if invalid_fields:
+                    raise ValueError(f"Invalid custom field(s): {', '.join(invalid_fields)}")
+
+            # Begin transaction
+            with session.begin_nested():
+                # Insert into tickets table
+                query = '''INSERT INTO tickets (chat_id, title, description, priority)
+                           VALUES (:chat_id, :title, :description, :priority)'''
+                session.execute(
+                    text(query),
+                    {
+                        "chat_id": chat_id,
+                        "title": title,
+                        "description": description,
+                        "priority": priority,
+                    },
+                )
+
+                # Fetch the generated ticket_id
+                result = session.execute("SELECT LAST_INSERT_ID()").fetchone()
```

Comment:
This function might give us a wrong ID, if there are concurrent requests
Lets make it SELECT FROM WHERE and get the ticket id

Reply:
Modified Sir

I wrote a condition combining the created_at field with the chat_id and other fields to ensure that we retrieve the specific ticket we just inserted.